contracts/interchain-token-service/src/contract.rs

```diff
@@ -271,40 +278,43 @@ impl InterchainTokenServiceInterface for InterchainTokenService {
         destination_chain: String,
         gas_token: Token,
     ) -> Result<BytesN<32>, ContractError> {
-        caller.require_auth();
-
         let deploy_salt = Self::interchain_token_deploy_salt(env, caller.clone(), salt);
-        let token_id = Self::interchain_token_id(env, Address::zero(env), deploy_salt);
-        let token_address = Self::token_address(env, token_id.clone());
-        let token = token::Client::new(env, &token_address);
-        let token_metadata = TokenMetadata {
-            name: token.name(),
-            decimal: token.decimals(),
-            symbol: token.symbol(),
-        };
-
-        assert_ok!(validate_token_metadata(token_metadata.clone()));
-
-        let message = Message::DeployInterchainToken(DeployInterchainToken {
-            token_id: token_id.clone(),
-            name: token_metadata.name.clone(),
-            symbol: token_metadata.symbol.clone(),
-            decimals: token_metadata.decimal as u8,
-            minter: None,
-        });
+        Self::deploy_remote_token(env, caller, deploy_salt, destination_chain, gas_token)
+    }
 
-        InterchainTokenDeploymentStartedEvent {
-            token_id: token_id.clone(),
-            token_address,
-            destination_chain: destination_chain.clone(),
-            name: token_metadata.name,
-            symbol: token_metadata.symbol,
-            decimals: token_metadata.decimal,
-            minter: None,
-        }
-        .emit(env);
+    /// Deploys a remote canonical token on a specified destination chain.
+    ///
+    /// This function computes a deployment salt and uses it to deploy a canonical
+    /// representation of a token on the remote chain. It retrieves the token metadata
+    /// from the token address and ensures the metadata is valid before initiating
+    /// the deployment.
+    ///
+    /// # Arguments
+    /// * `env` - Reference to the environment object.
+    /// * `token_address` - The address of the token to be deployed.
+    /// * `destination_chain` - The name of the destination chain where the token will be deployed.
+    /// * `gas_token` - The token used to pay for gas during the deployment.
+    ///
+    /// # Returns
+    /// Returns the token ID of the deployed token on the remote chain, or an error if the deployment fails.
+    ///
+    /// # Errors
+    /// Returns `ContractError` if the deployment fails or if token metadata is invalid.
+    fn deploy_remote_canonical_token(
+        env: &Env,
+        token_address: Address,
+        destination_chain: String,
+        gas_token: Token,
+    ) -> Result<BytesN<32>, ContractError> {
+        let deploy_salt = Self::canonical_token_deploy_salt(env, token_address);
 
-        Self::pay_gas_and_call_contract(env, caller, destination_chain, message, gas_token)?;
+        let token_id = Self::deploy_remote_token(
+            env,
+            env.current_contract_address(),
```

Comment:
Bad style, there's no caller here. Remove it as an arg from deploy_remote_token. Caller should be checked in the entrypoint that requires it instead of the helper function where it doesn't quite make sense

Reply:
discussed in slack. added back caller as sender.